app/models/application_record.rb

```diff
@@ -16,6 +16,15 @@ def self.needs_migration?
     ActiveRecord::Migration.check_pending!
   end
 
+  # Determine whether the given search term is possibly a Primary Key (it's numeric and less than 4 bytes)
+  def self.possibly_pk?(search_term) # could add optional arg for 4 byte vs 8 byte, if needed later
+    numeric = /[\d-]+/.match(search_term).try(:[], 0) == search_term
```

Comment:
I could be missing something but this is trying to see if the whole term is numeric right? I think it would be clearer to write this as:
```suggestion
    numeric = search_term =~ /\A\d+\z/
```

or maybe more concisely:
```ruby
  MAX_PK = 2_147_483_648 
  def self.possibly_pk?(search_term)
   search_term =~ /\A\d+\z/ && search_term.to_i < MAX_PK
  end
```

Reply:
Ah, that makes sense! The ClientSearch concern is quite complex; the current pattern is including negative numbers for some reason.

You could try DRYing it up if you have high confidence that it's not changing behavior. Personally I think leaving it unchanged also reasonable given that it is complex and a bit risky to change

Reply:
Thanks, Theron. I wouldn't say I have high confidence, so I do think I'd like to leave it unchanged if other reviewers are okay with that solution.